### PR TITLE
Feature/before dev

### PR DESCRIPTION
**Pull request by sanjoggit**
some jwt add and permission implemented in another way

### CODE REVIEW

File: src/interfaces/utils.js [hunk 1]
```diff
@@ -63,27 +63,99 @@ const localClient =	createApiClient({
 	userAgent: API_CLIENT_USER_AGENT
 });
 
-export function hasPermission(profile, user) {
-	const permitted = profile.auth.role.some(profileRole => {
+export function removeGroupPrefix(user) {
```
Comment (natlibfi-arlehiko) [inline src/interfaces/utils.js]: What's this? You shouldn't edit names but from groups to roles. The mapping need to defined in configuration. Either with environment variables (One variable per role) or a separate file.

Comment (sanjoggit) [reply src/interfaces/utils.js]: it was old code, it shouldn't be there, i will try to send new pull request again

Comment (sanjoggit) [reply src/interfaces/utils.js]: oh you mean removeGroupPrefix()? It removes the prefix identifier-services-test- from groups

Comment (natlibfi-arlehiko) [reply src/interfaces/utils.js]: Yeah, but they need to be mapped to roles.

Comment (sanjoggit) [reply src/interfaces/utils.js]: actually role doesn't have prefix added, so do we have to add prefix while creating user?

Comment (natlibfi-arlehiko) [reply src/interfaces/utils.js]: The authentication can have any group names but the application has predefined role names. The group names -> role mappings must be configurable.

So, the authentication provider might have a group named `foo` which we want to map to role `publisherAdmin`. Therefore the the should be a function that reads configuration and maps user group `foo` to role `publisherAdmin`.

File: src/api.json [hunk 2]
```diff
@@ -2770,6 +2770,11 @@
           "createdResource": {
             "description": "Identifier of the resource that was created based on this request",
             "type": "string"
+          },
+          "replyTo": {
+            "description": "Email address of the user which will be used to notify changes in his/her request",
+            "type": "string",
+            "format": "email"
```
Comment (natlibfi-arlehiko) [inline src/api.json]: - The property value is not a email address, but user id instead (User id might have a form of an email address, but not necessarily). The email address is fetched using the user id.
- Change property name to **creator**

Comment (Codefeak) [reply src/api.json]: I made the changes accordingly.

File: src/interfaces/interfaceModules.js [hunk 3]
```diff
@@ -52,11 +52,21 @@ export default function (collectionName) {
...
-		const doc = await db.collection(collectionName).findOne({
-			_id: new ObjectId(id)
-		}, {
-			projection: protectedProperties
-		});
+		let doc;
+		if (collectionName === 'userMetadata') {
+			const newId = /[0-9a-fA-F]{24}/.test(id) ? new ObjectId(id) : id;
+			doc = await db.collection(collectionName).findOne({$or: 
+				[{id: newId}, {_id: newId}]
```
Comment (natlibfi-arlehiko) [inline src/interfaces/interfaceModules.js]: Why do you search using `_id`. That shouldn't be used for anything.

**Pull request merged**